lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,435 @@
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format('flowframe', 'flowframe', FlowFrame)
+
+
+class FlowSOM(Binary):
+    """R Object containing fSOM saved with saveRDS"""
+    file_ext = 'fsom'
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary RDS fsom file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary RDS fsom (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """
+        Checking if the file is a FlowSOM R object.
+
+        For this to work, need to have install checkFlowSOM.R via bioconda
+        conda install ig-checkflowtypes
+        """
+        try:
+            rscript = 'checkFlowSOM.R'
+            fcs_check = subprocess.check_output([rscript, filename])
+            if re.search('TRUE', str(fcs_check)):
+                return True
+            else:
+                return False
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format('fsom', 'fsom', FlowSOM)
+
+
+class FlowSet(Binary):
+    """R Object containing flowSet saved with saveRDS"""
+    file_ext = 'flowset'
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary RDS flowSet file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary RDS flowSet (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """
+        Checking if the file is a flowSet R object.
+
+        For this to work, need to have install checkFlowSet.R via bioconda
+        conda install ig-checkflowtypes
+        """
+        try:
+            rscript = 'checkFlowSet.R'
+            fcs_check = subprocess.check_output([rscript, filename])
+            if re.search('TRUE', str(fcs_check)):
+                return True
+            else:
+                return False
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format('flowset', 'flowset', FlowSet)
```

Comment:
Shouldn't this be done via the sniffers section in datatype.conf?

Reply:
yes, this is done as well in datatype conf.xml, is this redundant then? This is the first time I mess so closely with datatypes, so I'm a bit ignorant on this.

Reply:
I removed all RData ones and just left them as subclass of Rdata in the datatype.conf.xml file, so this one in particular is gone, but there are others in the file (like the fcs one).

Reply:
I'm not sure